correlation_rules/github_repo_cloned_followed_by_aws_login.yml

```diff
@@ -0,0 +1,26 @@
+AnalysisType: correlation_rule
+RuleID: "GitHub.Repo.Cloned.Followed.By.AWS.Login"
+DisplayName: "GitHub Repo Cloned Followed By AWS Login"
+Enabled: true
+Severity: Medium
+Description: The threat actor appears to have used automated tools to continually clone public GitHub repositories and scan for exposed Amazon Web Services (AWS) IAM credentials. The threat actor also appeared to blocklist AWS accounts that routinely expose IAM credentials, in what we believe to be an effort to prevent security researchers from following their operations. The threat actors might have perceived them to be obvious honey traps.
+Reference: https://unit42.paloaltonetworks.com/malicious-operations-of-exposed-iam-keys-cryptojacking/
+Detection:
+    - Sequence:
+        - ID: ID.1
+          RuleID: GitHub.Repository.Cloned
+          MinMatchCount: 1
+        - ID: ID.2
+          RuleID: AWS.Console.Login
+          MinMatchCount: 1
+      Transitions:
+        - ID: TR1
+          From: ID.1
+          To: ID.2
+          WithinTimeFrameMinutes: 60
+          Match:
+            - On: p_alert_context.caller_ip
```

Comment:
This definitely works, using the alert context field to stick a value in a certain location to reference.

I wanted to add that we've just added the capability to match on arbitrary fields, so you could alternatively do:

```
Match:
  - From: p_udm.source.ip
    To: resource.actor_ip
```

And then get rid of the `AlertContext` definitions from the two signal rules. Either way works so ultimately it is a style choice, but I find referencing the fields here directly to require less mental load (as you don't have to go look up what the alert context function in the subrule is doing to understand it).

Reply:
the arbitrary fields have to be available in the schema (they are validated) and the rule being referenced can only have 1 schema associated with it, too.